src/test/java/teammates/storage/api/NotificationsDbTest.java

```diff
@@ -0,0 +1,359 @@
+package teammates.storage.api;
+
+import java.lang.reflect.Method;
+import java.time.Instant;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.UUID;
+
+import org.testng.annotations.AfterMethod;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import teammates.common.datatransfer.NotificationTargetUser;
+import teammates.common.datatransfer.attributes.NotificationAttributes;
+import teammates.common.exception.EntityAlreadyExistsException;
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.InvalidParametersException;
+import teammates.test.BaseTestCaseWithLocalDatabaseAccess;
+
+/**
+ * SUT: {@link NotificationsDb}.
+ */
+public class NotificationsDbTest extends BaseTestCaseWithLocalDatabaseAccess {
+
+    static final String NON_EXISTENT_ID = "invalid_notification_id";
+
+    private final NotificationsDb notificationsDb = NotificationsDb.inst();
+    private final Map<String, NotificationAttributes> typicalNotifications = getTypicalDataBundle().notifications;
+
+    // Notification attributes will be assigned with value at setup()
+    private NotificationAttributes n;
+    private Set<String> expectedIds = new HashSet<>();
+
+    @BeforeMethod
+    public void setup(Method method) throws Exception {
+        switch (method.getName()) {
+        case "testGetNotification":
+        case "testUpdateNotification":
+        case "testUpdateNotification_bulkUpdate_shouldUpdateSuccessfully":
+        case "testUpdateNotification_singleFieldUpdate_shouldUpdateSuccessfully":
+        case "testDeleteNotification":
+            n = notificationsDb.createEntity(typicalNotifications.get("notification1"));
+            break;
+        case "testGetAllNotifications":
+            for (NotificationAttributes n : typicalNotifications.values()) {
+                notificationsDb.createEntity(n);
+                expectedIds.add(n.getNotificationId());
+            }
+            break;
+        case "testGetActiveNotificationsByTargetUser":
+        case "testCreateNotification":
+        case "testHasExistingEntities":
+            // Do nothing - leave the initialization to the test method.
+            break;
+        default:
+            // All test methods should be covered above, so this should not be reached.
+            assert false;
+            break;
+        }
+    }
+
+    /**
+     * Removes all notifications created by each test.
+     */
+    @AfterMethod
+    public void cleanUp() {
+        List<NotificationAttributes> retrieved = notificationsDb.getAllNotifications();
+        assertNotNull(retrieved);
+
+        ______TS("clean up: delete " + retrieved.size() + " notifications");
+        retrieved.forEach(n -> notificationsDb.deleteNotification(n.getNotificationId()));
```

Comment:
I don't think this can be considered a test case. 
`______TS` is a Test Segment divider. It is used to divide a test case into logical sections.

Reply:
Thanks for the comment. As this is the clean up I think I can remove it.